cli/js/testing.ts

```diff
@@ -120,70 +121,106 @@ export function test(
   TEST_REGISTRY.push(testDef);
 }
 
-interface TestStats {
-  filtered: number;
-  ignored: number;
-  measured: number;
-  passed: number;
-  failed: number;
+export interface RunTestsStartMessage {
+  kind: "runTestsStart";
+  tests: TestDefinition[];
 }
 
-export interface RunTestsOptions {
-  exitOnFail?: boolean;
-  failFast?: boolean;
-  only?: string | RegExp;
-  skip?: string | RegExp;
-  disableLog?: boolean;
-  reporter?: TestReporter;
+export interface TestStartMessage extends TestDefinition {
+  kind: "testStart";
 }
 
-enum TestStatus {
-  Passed = "passed",
-  Failed = "failed",
-  Ignored = "ignored"
-}
-
-interface TestResult {
+export interface TestEndMessage {
+  kind: "testEnd";
   name: string;
-  status: TestStatus;
+  status: "passed" | "failed" | "ignored";
   duration: number;
   error?: Error;
 }
 
-export enum TestEvent {
-  Start = "start",
-  TestStart = "testStart",
-  TestEnd = "testEnd",
-  End = "end"
+export interface RunTestsEndMessage {
+  kind: "runTestsEnd";
+  filtered: number;
+  ignored: number;
+  measured: number;
+  passed: number;
+  failed: number;
+  duration: number;
+  errors: Array<[string, Error]>;
 }
 
-interface TestEventStart {
-  kind: TestEvent.Start;
-  tests: number;
-}
+export type TestMessage =
+  | RunTestsStartMessage
+  | TestStartMessage
+  | TestEndMessage
+  | RunTestsEndMessage;
```

Comment:
Maybe change it to:
```
export interface TestMessage = {
   start?: {
     ...
   },
   testStart?: {},
   testEnd?: {},
   end?: {}
}
```
That will give a single exported interface instead of 5 types

Reply:
Nice! Works great with index types: https://github.com/denoland/deno/blob/beccd4f0d91cb17391c8cddff20614367bb0be80/cli/js/lib.deno.ns.d.ts#L83